experimental/list_formatter/src/list_formatter.rs

```diff
@@ -105,12 +70,23 @@ pub mod parts {
 
 /// The [`Writeable`] implementation that is returned by [`ListFormatter::format`]. See
 /// the [`writeable`] crate for how to consume this.
-pub struct List<'a, W: Writeable + 'a, I: Iterator<Item = W> + Clone + 'a> {
-    formatter: &'a ListFormatter,
+pub struct List<
+    'a,
+    M: ResourceMarker<Yokeable = ListFormatterPatternsV1<'static>>,
```

Comment:
Note: If you make `M` infectious, there's going to be a lot of places we have to write this bound.